public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsQueryEditor.tsx

```diff
@@ -72,21 +76,49 @@ const LogsQueryEditor = ({
 
   useEffect(() => {
     if (query.azureLogAnalytics?.resources && query.azureLogAnalytics.resources.length === 1) {
-      const resource = parseResourceURI(query.azureLogAnalytics.resources[0]);
+      const selectedResource = templateSrv?.replace(query.azureLogAnalytics.resources[0]);
+      const parsedResource = parseResourceURI(selectedResource);
       setShowBasicLogsToggle(
-        resource.metricNamespace?.toLowerCase() === 'microsoft.operationalinsights/workspaces' && basicLogsEnabled
+        parsedResource.metricNamespace?.toLowerCase() === 'microsoft.operationalinsights/workspaces' && basicLogsEnabled
       );
     } else {
       setShowBasicLogsToggle(false);
     }
-  }, [basicLogsEnabled, query.azureLogAnalytics?.resources]);
+  }, [basicLogsEnabled, query.azureLogAnalytics?.resources, templateSrv]);
 
   useEffect(() => {
     if ((!basicLogsEnabled || !showBasicLogsToggle) && query.azureLogAnalytics?.basicLogsQuery) {
       const updatedBasicLogsQuery = setBasicLogsQuery(query, false);
       onChange(setKustoQuery(updatedBasicLogsQuery, ''));
     }
   }, [basicLogsEnabled, onChange, query, showBasicLogsToggle]);
+
+  useEffect(() => {
+    if (showBasicLogsToggle && query.azureLogAnalytics?.basicLogsQuery && !!query.azureLogAnalytics.query) {
+      const querySplit = query.azureLogAnalytics.query.split('|');
+      // Basic Logs queries are required to start the query with a table
+      const table = querySplit[0].trim();
+      datasource.azureLogAnalyticsDatasource.getBasicLogsQueryUsage(query, table).then((data) => setDataIngested(data));
```

Comment:
Prefer using async/await rather than callbacks

Reply:
changed this :)